timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionTestSetup.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.corruption.detection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.SetMultimap;
+import com.palantir.paxos.Client;
+import com.palantir.paxos.ImmutableNamespaceAndUseCase;
+import com.palantir.paxos.NamespaceAndUseCase;
+import com.palantir.paxos.PaxosAcceptorState;
+import com.palantir.paxos.PaxosStateLog;
+import com.palantir.paxos.PaxosValue;
+import com.palantir.paxos.SqliteConnections;
+import com.palantir.paxos.SqlitePaxosStateLog;
+import com.palantir.timelock.history.LocalHistoryLoader;
+import com.palantir.timelock.history.PaxosLogHistoryProvider;
+import com.palantir.timelock.history.TimeLockPaxosHistoryProvider;
+import com.palantir.timelock.history.models.AcceptorUseCase;
+import com.palantir.timelock.history.models.CompletePaxosHistoryForNamespaceAndUseCase;
+import com.palantir.timelock.history.models.LearnerUseCase;
+import com.palantir.timelock.history.remote.TimeLockPaxosHistoryProviderResource;
+import com.palantir.timelock.history.sqlite.SqlitePaxosStateLogHistory;
+import com.palantir.timelock.history.utils.PaxosSerializationTestUtils;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.sql.DataSource;
+import org.immutables.value.Value;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+public abstract class TimeLockCorruptionTestSetup {
```

Comment:
chatted offline - I personally don't like this structure of using abstract classes; would prefer if this was a final class that you call methods on directly.